core/src/main/scala/pickling/Pickler.scala

```diff
@@ -26,6 +27,7 @@ trait SPickler[T] {
 
 object SPickler {
   def generate[T]: SPickler[T] = macro Compat.PicklerMacros_impl[T]
+  implicit def spuToSp[T: SPicklerUnpickler]: SPickler[T] = implicitly[SPicklerUnpickler[T]].pickler
```

Comment:
There is 1-less implicit lookup with @havocp's approach, but Eugene's gives us an additional degree of freedom w/ implicit magickery.

I agree you should just never have two picklers in scope (ideally) or it's buggy code.

Also, the final class should be a "SpicklerUnpicklerImpl" class, while the trait could be abstract for custom-written Pickller/Unpickler code.